src/api/NotesApi.ts

```diff
@@ -84,6 +67,43 @@ export class NotesApi implements INotesApi {
         }
     }
 
+    async fetchAllNotes(requestIds: number[]): Promise<INote[]> {
+        try {
+            let batch = sp.web.createBatch();
+            let promises: Promise<SPNote[]>[] = requestIds.map(id => this.notesList.items.select("Id", "Request/Id", "Title", "Text", "Modified", "Author/Id", "Author/Title", "Author/EMail", "Status").filter(`RequestId eq ${id}`).expand("Request", "Author").inBatch(batch).get());
+            batch.execute();
+
+            let notes: SPNote[] = [];
+            for (let p of promises) {
+                (await p).forEach(note => notes.push(note));
+            }
```

Comment:
Not sure it really matters, but should the await be on the batch.execute() rather than the promises?